### DISCUSSION TITLE
I want to create my own vcpkg from dlls, libs, and include folders

### DISCUSSION THREAD

**Discussion by smbika007** [microsoft/vcpkg#26814] [Q&A]
Hi, 

I have dlls, libs, and includes for a project that has no vcpkg available for download and install. I need these dependencies for a larger project that relies on "findable" vcpkgs in its CMakelists.txt files for handling 3rd party dependencies. 

I created a repo on GitHub which has folders:

include (the include files and subfolders)
lib (.lib and .dll files)
licenses (license file)

Can I and, if so, how do I create my own vcpkg to use in my CMakelists.txt file?

Thanks,
Sean

**Comment by JackBoosY** [comment]
So, you request is to add a completely custom port right?
If yes, use overlay-ports, install all your binaries / headers / license in `portfile.cmake`, and use `find_path` / `find_library` in your another custom project's CMakeLists.

Also, you can write a simply `Find<PORT_NAME>.cmake` which contains `find_path` and `find_library` then install it into _${CURRENT_PACKAGES_DIR}/share/${PORT}_, and write a simply `vcpkg-cmake-wrapper.cmake` which set _${CURRENT_PACKAGES_DIR}/share/${PORT}_ to `CMAKE_PREFIX_PATH` before call `find_package(${ARGS})`, so you can directly use `find_package` in your custom project's CMakeLists. You can find the sample code in this repo.
